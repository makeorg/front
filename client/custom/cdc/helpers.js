import { bretagneQuestionSlugs } from './constants';

export const getBretagneQuestionSlug = (slug: string) =>
  bretagneQuestionSlugs.find(
    bretagneQuestionSlug => bretagneQuestionSlug === slug
  );
