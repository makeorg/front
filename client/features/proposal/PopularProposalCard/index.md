##### PopularProposalCard Example :

```jsx
import { proposalTypeFixture } from '../../../../shared/types/__fixtures__/proposal.fixture.js';
import { PopularProposalCard } from './index';

<ol
  style={{
    width: '100% ',
    height: '440px',
    listStyle: 'none',
    padding: '15px',
    background: '#f2f2f2',
  }}
>
  <PopularProposalCard proposal={proposalTypeFixture} />
</ol>;
```
