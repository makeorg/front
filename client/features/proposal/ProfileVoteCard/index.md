##### ProfileVoteCard Example :
```jsx
import { proposalTypeFixture } from '../../../../shared/types/__fixtures__/proposal.fixture.js';
import { organisationTypeFixture } from '../../../../shared/types/__fixtures__/organisation.fixture.js';
import { ProfileVoteCard } from './index';

<ul style={{listStyle: 'none', padding: '15px', background: '#f2f2f2'}}>
  <ProfileVoteCard 
    voteKey='agree'
    proposal={proposalTypeFixture}
    organisation={organisationTypeFixture}
  />
</ul>
```