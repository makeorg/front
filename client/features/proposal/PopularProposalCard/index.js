// @flow
import React from 'react';
import { i18n } from 'Shared/i18n';
import { isInProgress } from 'Shared/helpers/date';
import { type ProposalType } from 'Shared/types/proposal';
import {
  ProposalStyle,
  ProposalInnerStyle,
} from 'Client/ui/Elements/ProposalCardElements';
import { ProposalAuthorElement } from 'Client/ui/Proposal/AuthorElement';
import { AuthorWrapperStyle } from 'Client/ui/Proposal/AuthorElement/Styled';
import { ScreenReaderItemStyle } from 'Client/ui/Elements/AccessibilityElements';
import { Vote } from 'Client/features/vote';
import { DetailledVoteResults } from 'Client/ui/Proposal/DetailledVoteResults';
import { getProposalLink } from 'Shared/helpers/url';
import { useSelector } from 'react-redux';
import { type StateRoot } from 'Shared/store/types';
import {
  PopularProposalWrapperStyle,
  PopularProposalHeader,
  PopularProposalTagStyle,
} from './style';

type Props = {
  /** Object with all proposal's properties */
  proposal: ProposalType,
  /** Proposal's position in list */
  position: number,
  /** Size of proposals list */
  size: number,
};

export const PopularProposalCard = ({
  proposal,
  position = 0,
  size = 0,
}: Props) => {
  const { country } = useSelector((state: StateRoot) => state.appConfig);
  const canVote = isInProgress(proposal.question);
  return (
    <PopularProposalWrapperStyle aria-posinset={position} aria-setsize={size}>
      <PopularProposalHeader aria-hidden>
        {`#${position}`}
      </PopularProposalHeader>
      <AuthorWrapperStyle>
        <ProposalAuthorElement proposal={proposal} withAvatar />
      </AuthorWrapperStyle>
      <ProposalInnerStyle>
        <ScreenReaderItemStyle>
          {i18n.t('proposal_card.content')}
        </ScreenReaderItemStyle>
        <ProposalStyle
          id={`proposal_content_${position}`}
          to={getProposalLink(
            country,
            proposal.slug,
            proposal.id,
            proposal.slug
          )}
        >
          {proposal.content}
        </ProposalStyle>
        {canVote ? (
          <Vote
            proposal={proposal}
            votes={proposal.votes}
            proposalKey={proposal.proposalKey}
            index={position}
          />
        ) : (
          <DetailledVoteResults
            votes={proposal.votes}
            proposalId={proposal.id}
          />
        )}
      </ProposalInnerStyle>
      {proposal.selectedStakeTag && proposal.selectedStakeTag.display && (
        <PopularProposalTagStyle>
          <ScreenReaderItemStyle
            dangerouslySetInnerHTML={{
              __html: i18n.t('consultation.tags.proposal_list'),
            }}
          />
          {proposal.selectedStakeTag.label}
        </PopularProposalTagStyle>
      )}
    </PopularProposalWrapperStyle>
  );
};
