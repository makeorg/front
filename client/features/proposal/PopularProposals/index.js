// @flow
import React, { useEffect, useRef } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { TileWithTitle } from 'Client/ui/Elements/TileWithTitle';
import { type Question } from 'Shared/types/question';
import { type TypeSliderParams } from 'Shared/types/views';
import { i18n } from 'Shared/i18n';
import { fetchPopularProposals } from 'Shared/store/actions/question';
import {
  type StateRoot,
  type PopularProposals as PopularProposalsType,
} from 'Shared/store/types';
import { PopularProposalCard } from 'Client/features/proposal/PopularProposalCard';
import { useSlider } from 'Client/hooks/useSlider';
import { Breakpoints } from 'Client/app/assets/vars/Breakpoints';
import { SvgArrowLeft, SvgArrowRight } from 'Client/ui/Svg/elements';
import { FlexElementStyle } from 'Client/ui/Elements/FlexElements';
import { HiddenItemStyle } from 'Client/ui/Elements/HiddenElements';
import { GliderStylesheet } from 'Client/app/assets/css-in-js/GliderStyle';
import { ProposalsSliderWrapperStyle } from 'Client/features/homepage/Proposals/Styled';
import { BackgroundColors } from 'Client/app/assets/vars/Colors';
import { useMobile } from 'Client/hooks/useMedia';
import {
  PopularProposalsSliderListItemStyle,
  PopularProposalsSliderListStyle,
  ArrowStyle,
} from './style';

type Props = {
  question: Question,
};

const sliderName = 'popularProposals';
const sliderParams: TypeSliderParams = {
  slidesToShow: 1.5,
  responsive: [
    {
      breakpoint: Breakpoints.Tablet,
      settings: {
        slidesToShow: 2.5,
      },
    },
  ],
  draggable: true,
  interactiveChildren: {
    links: true,
    buttons: true,
  },
  arrows: {
    prev: `.${sliderName}.glider-prev`,
    next: `.${sliderName}.glider-next`,
  },
};

export const PopularProposals = ({ question }: Props) => {
  const sliderRef = useRef();
  const isMobile = useMobile();
  const dispatch = useDispatch();
  const proposals: PopularProposalsType = useSelector(
    (state: StateRoot) => state.questions[question.slug].popularProposals
  );

  useEffect(() => {
    dispatch(fetchPopularProposals(question.questionId, question.slug));
  }, [dispatch, question.questionId]);

  const hasProposals = proposals && proposals.results.length > 0;

  useSlider(sliderRef, sliderParams, hasProposals);

  if (!hasProposals) {
    return null;
  }

  return (
    <TileWithTitle
      title={i18n.t('consultation.popular_proposals.title', {
        count: proposals.results.length,
      })}
      style={{
        backgroundColor: BackgroundColors.ExtraLightGrey,
        marginTop: '15px',
      }}
    >
      <ProposalsSliderWrapperStyle className={`${sliderName} glider-contain`}>
        <GliderStylesheet />
        <FlexElementStyle as={isMobile ? HiddenItemStyle : FlexElementStyle}>
          <ArrowStyle
            className={`${sliderName} glider-prev`}
            aria-label={i18n.t('common.slider.previous')}
            aria-controls="glide_translator"
            direction="left"
          >
            <SvgArrowLeft aria-hidden />
          </ArrowStyle>
          <ArrowStyle
            className={`${sliderName} glider-next`}
            aria-label={i18n.t('common.slider.next')}
            aria-controls="glide_translator"
            direction="right"
          >
            <SvgArrowRight aria-hidden />
          </ArrowStyle>
        </FlexElementStyle>
        <div className={`${sliderName} glider`} ref={sliderRef}>
          <PopularProposalsSliderListStyle
            className={`${sliderName} glider-track`}
          >
            {proposals.results.map((proposal, index) => (
              <PopularProposalsSliderListItemStyle
                key={proposal.id}
                className={sliderName}
              >
                <PopularProposalCard
                  proposal={proposal}
                  position={index + 1}
                  size={proposals.results.length}
                />
              </PopularProposalsSliderListItemStyle>
            ))}
          </PopularProposalsSliderListStyle>
        </div>
      </ProposalsSliderWrapperStyle>
    </TileWithTitle>
  );
};
