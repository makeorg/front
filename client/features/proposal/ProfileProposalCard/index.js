// @flow
import React from 'react';
import { type Proposal as TypeProposal } from 'Shared/types/proposal';
import { getProposalLink, getConsultationLink } from 'Shared/helpers/url';
import { DetailledVoteResults } from 'Client/ui/Proposal/DetailledVoteResults';
import { ProposalAuthorElement } from 'Client/ui/Proposal/AuthorElement';
import { ProposalFooterWithQuestionElement } from 'Client/ui/Proposal/FooterElement';
import { ProfileProposalCardStyle, ProposalStyle } from './Styled';

type Props = {
  /** Object with all proposal's properties */
  proposal: TypeProposal,
  /** Proposal's position in list */
  position: number,
  /** Size of proposals list */
  size: number,
  /** Optional display for proposal's status */
  withStatus?: boolean,
};

export const ProfileProposalCard = (props: Props) => {
  const { proposal, position, size } = props;
  const { author, question } = proposal;
  const formattedProposalStatus = proposal.status.toLowerCase();
  const isProposalAccepted = formattedProposalStatus === 'accepted';

  return (
    <ProfileProposalCardStyle
      role="article"
      aria-posinset={position}
      aria-setsize={size}
      className={`proposal-${formattedProposalStatus}`}
    >
      <ProposalAuthorElement
        author={author}
        country={proposal.country}
        language={proposal.language}
        createdAt={proposal.createdAt}
        withAvatar
        formattedProposalStatus={formattedProposalStatus}
      />
      <ProposalStyle
        id={`proposal_content_${position}`}
        {...(isProposalAccepted
          ? {
              href: getProposalLink(
                proposal.country,
                proposal.language,
                question.slug,
                proposal.id,
                proposal.slug
              ),
            }
          : { as: 'p' })}
      >
        {proposal.content}
      </ProposalStyle>
      {isProposalAccepted && (
        <DetailledVoteResults votes={proposal.votes} proposalId={proposal.id} />
      )}
      <ProposalFooterWithQuestionElement
        question={question}
        consultationLink={getConsultationLink(
          proposal.country,
          proposal.language,
          question.slug
        )}
        isProposalAccepted={isProposalAccepted}
      />
    </ProfileProposalCardStyle>
  );
};

ProfileProposalCard.defaultPropTypes = {
  withStatus: false,
};
