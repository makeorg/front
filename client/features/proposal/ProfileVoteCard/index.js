// @flow

import React from 'react';
import { i18n } from 'Shared/i18n';
import { type Proposal as TypeProposal } from 'Shared/types/proposal';
import { type Organisation as TypeOrganisation } from 'Shared/types/organisation';
import {
  getProposalLink,
  getOrganisationProfileLink,
} from 'Shared/helpers/url';
import { voteStaticParams } from 'Shared/constants/vote';
import { RedLinkStyle } from 'Client/ui/Elements/LinkElements';
import { SvgCheckedSymbol, SvgThumbsUp } from 'Client/ui/Svg/elements';
import { VoteResultElement } from 'Client/ui/Proposal/VoteResultElement';
import { ProposalFooterWithTagElement } from 'Client/ui/Proposal/FooterElement/ProposalWithTag';
import { ProposalAuthorElement } from 'Client/ui/Proposal/AuthorElement';
import { TextColors } from 'Client/app/assets/vars/Colors';
import {
  ProposalCardStyle,
  ProposalStyle,
  ProposalInnerStyle,
} from 'Client/ui/Elements/ProposalCardElements';
import { AuthorWrapperStyle } from 'Client/ui/Proposal/AuthorElement/Styled';
import { ButtonIconWrapperStyle } from 'Client/ui/Elements/Vote/Styled';
import {
  ProfileVoteCardStyle,
  ProfileVoteWrapperStyle,
  ProfileVoteTitleStyle,
  ProfileVoteDescriptionStyle,
  ProfileHasVotedStyle,
} from './Styled';

type Props = {
  /** Object with all organisation's properties */
  organisation: TypeOrganisation,
  /** Object with all proposal's properties */
  proposal: TypeProposal,
  /** Nature of the organisation's vote */
  voteKey: string,
  /** Proposal's position in list */
  position: number,
  /** Size of proposals list */
  size: number,
};

export const ProfileVoteCard = ({
  voteKey,
  organisation,
  proposal,
  position,
  size,
}: Props) => {
  const { author, question } = proposal;
  const voteAttributes = voteStaticParams[voteKey];
  return (
    <ProfileVoteWrapperStyle aria-posinset={position} aria-setsize={size}>
      <ProfileVoteTitleStyle>
        <ProfileHasVotedStyle
          aria-label={voteAttributes.label}
          color={voteAttributes.color}
        >
          <ButtonIconWrapperStyle transform={voteAttributes.transform}>
            <SvgThumbsUp aria-hidden />
          </ButtonIconWrapperStyle>
        </ProfileHasVotedStyle>
        <ProfileVoteDescriptionStyle>
          <RedLinkStyle
            href={getOrganisationProfileLink(
              organisation.country,
              organisation.language,
              organisation.slug
            )}
          >
            {organisation.organisationName}
          </RedLinkStyle>
          &nbsp;
          <SvgCheckedSymbol
            style={{ fontSize: '14px', fill: TextColors.Blue }}
          />
          &nbsp;
          {i18n.t(`profile.organisation.proposal_${voteKey}`)}
        </ProfileVoteDescriptionStyle>
      </ProfileVoteTitleStyle>

      <ProfileVoteCardStyle>
        <ProposalCardStyle as="div">
          <AuthorWrapperStyle>
            <ProposalAuthorElement
              author={author}
              country={proposal.country}
              language={proposal.language}
              createdAt={proposal.createdAt}
              withAvatar
            />
          </AuthorWrapperStyle>
          <ProposalInnerStyle>
            <ProposalStyle
              id={`proposal_content_${position}`}
              href={getProposalLink(
                proposal.country,
                proposal.language,
                question.slug,
                proposal.id,
                proposal.slug
              )}
            >
              {proposal.content}
            </ProposalStyle>
            <VoteResultElement
              proposalId={proposal.id}
              votes={proposal.votes}
              votedKey={voteKey}
            />
          </ProposalInnerStyle>

          <ProposalFooterWithTagElement tags={proposal.tags} />
        </ProposalCardStyle>
      </ProfileVoteCardStyle>
    </ProfileVoteWrapperStyle>
  );
};
