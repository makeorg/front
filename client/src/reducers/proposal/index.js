/* @flow */

import * as actionTypes from 'Constants/actionTypes';
import { getBaitText } from 'Constants/proposal';

const initialState: Object = {
  isTyping: false,
  canSubmit: false,
  hasProposed: false,
  isCurrentSubmitSuccess: false,
  content: '',
  length: getBaitText().length,
  questionId: null,
  error: null
};

export default function proposal(state: Object = initialState, action: Object) {
  switch (action.type) {
    case actionTypes.PROPOSE_TYPING:
      return {
        ...state,
        isTyping: true,
        isCurrentSubmitSuccess: false,
        content: action.content,
        length: action.length,
        canSubmit: action.canSubmit
      };
    case actionTypes.PROPOSE_REQUEST:
      return {
        ...state,
        isTyping: false,
        questionId: action.questionId
      };
    case actionTypes.PROPOSE_SUCCESS:
      return {
        ...state,
        isTyping: false,
        canSubmit: false,
        isCurrentSubmitSuccess: true,
        content: '',
        length: getBaitText().length,
        hasProposed: true,
        questionId: null,
        error: null
      };
    case actionTypes.PROPOSE_FAILURE:
      return {
        ...state,
        error: action.error
      };
    case actionTypes.PROPOSAL_LOAD:
      return {
        ...state,
        proposal: action.proposal
      };
    default:
      return state;
  }
}
