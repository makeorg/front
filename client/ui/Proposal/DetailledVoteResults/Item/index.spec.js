import React from 'react';
import renderer from 'react-test-renderer';
import { proposalTypeFixture } from 'Shared/types/__fixtures__/proposal.fixture';
import { getTotalVotesCount, getVotesPercent } from 'Shared/helpers/voteResult';
import { DetailledResultItem } from './index';

jest.mock('Client/ui/Elements/AccessibilityElements', () => ({
  ScreenReaderItemStyle: 'ScreenReaderItemStyle',
}));

jest.mock('../Styled/Progress', () => ({
  DetailledItemStyle: 'DetailledItemStyle',
  VoteDataListStyle: 'VoteDataListStyle',
  VoteDataBoldItemStyle: 'VoteDataBoldItemStyle',
  VoteDataItemStyle: 'VoteDataItemStyle',
  QualificationDataListStyle: 'QualificationDataListStyle',
}));

jest.mock('Client/ui/Elements/Vote/Styled', () => ({
  IsVotedButtonStyle: 'IsVotedButtonStyle',
  ButtonIconWrapperStyle: 'ButtonIconWrapperStyle',
}));

const totalVotesCount = getTotalVotesCount(proposalTypeFixture.votes);
const votesPercent = getVotesPercent(
  proposalTypeFixture.votes,
  totalVotesCount
);

describe('DetailledResultItem', () => {
  const defaultProps = {
    vote: proposalTypeFixture.votes[0],
    votePercent: votesPercent[proposalTypeFixture.votes[0].voteKey],
  };

  it('must match the snapshot with default Props', () => {
    const component = renderer
      .create(<DetailledResultItem {...defaultProps} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
