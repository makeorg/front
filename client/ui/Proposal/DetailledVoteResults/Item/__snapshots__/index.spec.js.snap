// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailledResultItem must match the snapshot with default Props 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 {
  font-size: 12px;
  line-height: 22px;
  font-family: 'Roboto Bold',Arial,sans-serif;
  color: rgb(85,131,34);
}

.c8 {
  font-size: 12px;
  line-height: 22px;
  color: rgb(102,102,102);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0 15px;
  min-width: 200px;
}

.c0.disagree {
  border-left: 1px solid rgb(229,229,229);
  border-right: 1px solid rgb(229,229,229);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  margin-left: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  font-size: 12px;
  color: rgb(102,102,102);
  margin: 3px 0;
}

.c3 {
  font-size: 12px;
  color: rgb(102,102,102);
  margin: 3px 0;
  font-size: 14px;
  font-family: 'Roboto Bold',Arial,sans-serif;
}

.c5 {
  padding: 0;
  list-style: none;
  margin-top: 10px;
}

@media (min-width:767px) {
  .c7 {
    font-size: 14px;
  }
}

@media (min-width:767px) {
  .c8 {
    font-size: 14px;
  }
}

@media (min-width:767px) {
  .c4 {
    font-size: 14px;
  }
}

@media (min-width:767px) {
  .c3 {
    font-size: 14px;
  }
}

@media (min-width:767px) {
  .c3 {
    font-size: 16px;
  }
}

<li
  className="agree c0"
>
  <div
    className="c1"
  >
    <ScreenReaderItemStyle>
      vote.agree
    </ScreenReaderItemStyle>
    <IsVotedButtonStyle
      aria-hidden={true}
      as="div"
      color="rgb(85,131,34)"
    >
      <svg
        height="1em"
        viewBox="0 0 15 16"
        width="1em"
      >
        <path
          d="M14.522 8.942c.319-.474.478-.996.478-1.567 0-.66-.247-1.234-.742-1.721-.495-.488-1.08-.731-1.758-.731h-1.719c.313-.635.47-1.25.47-1.846 0-.75-.115-1.346-.343-1.789a2.13 2.13 0 0 0-.996-.976A3.38 3.38 0 0 0 8.438 0c-.333 0-.626.119-.88.356-.28.269-.481.615-.605 1.038-.124.423-.223.829-.298 1.217-.074.387-.19.661-.346.822-.32.34-.668.75-1.045 1.23-.658.84-1.104 1.337-1.338 1.49H1.25c-.345 0-.64.121-.884.361S0 7.044 0 7.384v6.154c0 .34.122.63.366.87s.539.361.884.361h2.812c.144 0 .593.128 1.348.385.8.275 1.506.485 2.114.63A7.998 7.998 0 0 0 9.375 16h1.26c.918 0 1.657-.258 2.217-.774s.836-1.22.83-2.11a2.68 2.68 0 0 0 .586-1.712c0-.141-.01-.279-.03-.414.248-.429.372-.89.372-1.384 0-.231-.03-.452-.088-.664zM2.315 13.356a.606.606 0 0 1-.44.183.606.606 0 0 1-.44-.183.587.587 0 0 1-.185-.433c0-.167.062-.31.186-.433a.606.606 0 0 1 .44-.182c.168 0 .315.06.439.182a.587.587 0 0 1 .185.433.587.587 0 0 1-.185.433zM13.54 8.164c-.14.295-.314.445-.522.452.098.109.179.261.244.457.065.195.098.373.098.533 0 .442-.173.824-.518 1.144.117.205.176.426.176.664a1.6 1.6 0 0 1-.17.706 1.078 1.078 0 0 1-.465.505c.033.192.049.372.049.538 0 1.07-.625 1.606-1.875 1.606H9.375c-.853 0-1.966-.234-3.34-.702a5.553 5.553 0 0 0-1.343-.437 4.48 4.48 0 0 0-.322-.063 1.999 1.999 0 0 0-.307-.029H3.75V7.385h.313a.85.85 0 0 0 .346-.087c.127-.057.258-.144.391-.26a5.657 5.657 0 0 0 .766-.764 19.1 19.1 0 0 0 .645-.803l.225-.288c.358-.436.609-.727.752-.875.267-.276.46-.626.58-1.053.121-.426.22-.828.298-1.206.079-.379.202-.651.372-.818.625 0 1.042.15 1.25.452.208.301.312.766.312 1.394 0 .378-.156.893-.469 1.543-.312.651-.468 1.162-.468 1.534H12.5c.326 0 .616.123.87.37.253.247.38.534.38.86 0 .225-.07.485-.21.78z"
        />
      </svg>
    </IsVotedButtonStyle>
    <div
      className="c2"
    >
      <p
        className="c3"
      >
        common.percent
      </p>
      <p
        className="c4"
      >
        vote.label
      </p>
    </div>
  </div>
  <ScreenReaderItemStyle>
    qualification.static_repartition
  </ScreenReaderItemStyle>
  <ul
    className="c5"
  >
    <li
      className="c6"
    >
      <span
        className="c7"
        color="rgb(85,131,34)"
      >
        qualification.likeIt
      </span>
      <ScreenReaderItemStyle>
         : 
      </ScreenReaderItemStyle>
      <span
        className="c8"
      >
         4 
      </span>
      <ScreenReaderItemStyle>
        qualification.times
      </ScreenReaderItemStyle>
    </li>
    <li
      className="c6"
    >
      <span
        className="c7"
        color="rgb(85,131,34)"
      >
        qualification.doable
      </span>
      <ScreenReaderItemStyle>
         : 
      </ScreenReaderItemStyle>
      <span
        className="c8"
      >
         2 
      </span>
      <ScreenReaderItemStyle>
        qualification.times
      </ScreenReaderItemStyle>
    </li>
    <li
      className="c6"
    >
      <span
        className="c7"
        color="rgb(85,131,34)"
      >
        qualification.platitudeAgree
      </span>
      <ScreenReaderItemStyle>
         : 
      </ScreenReaderItemStyle>
      <span
        className="c8"
      >
         2 
      </span>
      <ScreenReaderItemStyle>
        qualification.times
      </ScreenReaderItemStyle>
    </li>
  </ul>
</li>
`;
